client/packages/invoices/src/Prescriptions/DetailView/PrescriptionLineEdit/PrescriptionLineEdit.tsx

```diff
@@ -0,0 +1,282 @@
+        input.focus();
+      }
+    }, 500);
+  }, [batch]);
+};
+
+export const PrescriptionLineEdit: React.FC<ItemDetailsModalProps> = ({
+  isOpen,
+  onClose,
+  draft,
+  mode,
+}) => {
+  const item = !draft ? null : draft.item ?? null;
+  const t = useTranslation(['dispensary']);
+  const { info, warning } = useNotification();
+  const { Modal } = useDialog({ isOpen, onClose, disableBackdrop: true });
+  const [currentItem, setCurrentItem] = useBufferState(item);
+  const [isAutoAllocated, setIsAutoAllocated] = useState(false);
+
+  const { mutateAsync } = usePrescription.line.save();
+  const { status } = usePrescription.document.fields('status');
+  const isDisabled = usePrescription.utils.isDisabled();
+  const {
+    draftStockOutLines,
+    updateQuantity,
+    setDraftStockOutLines,
+    isLoading,
+    updateNotes,
+  } = useDraftPrescriptionLines(currentItem);
+  const packSizeController = usePackSizeController(draftStockOutLines);
+  const { next, disabled: nextDisabled } = useNextItem(currentItem?.id);
+  const { isDirty, setIsDirty } = useDirtyCheck();
+  const height = useKeyboardHeightAdjustment(700);
+  useFocusNumberOfPacksInput(draft);
+
+  const placeholder = draftStockOutLines?.find(
+    ({ type, numberOfPacks }) =>
+      type === InvoiceLineNodeType.UnallocatedStock && numberOfPacks !== 0
+  );
+
+  const onUpdateQuantity = (batchId: string, quantity: number) => {
+    updateQuantity(batchId, quantity);
+    setIsAutoAllocated(false);
+  };
+
+  const onUpdateNotes = (note: string) => {
+    updateNotes(note);
+    setIsAutoAllocated(false);
+  };
+
+  const onSave = async () => {
+    if (!isDirty) return;
+    console.log('onSave', draftStockOutLines);
+
+    await mutateAsync(draftStockOutLines);
+    if (!draft) return;
+  };
+
+  const onNext = async () => {
+    await onSave();
+    if (!!placeholder) {
+      const infoSnack = info(t('message.placeholder-line'));
+      infoSnack();
+    }
+    if (mode === ModalMode.Update && next) setCurrentItem(next);
+    else if (mode === ModalMode.Create) setCurrentItem(null);
+    else onClose();
+    setIsDirty(false);
+    // Returning true here triggers the slide animation
+    return true;
+  };
+
+  const onAllocate = (
+    newVal: number,
+    packSize: number | null,
+    autoAllocated = false
+  ) => {
+    const newAllocateQuantities = allocateQuantities(
+      status,
+      draftStockOutLines
+    )(newVal, packSize);
+    setIsDirty(true);
+    setDraftStockOutLines(newAllocateQuantities ?? draftStockOutLines);
+    setIsAutoAllocated(autoAllocated);
+
+    const newAllocatedTotal = newAllocateQuantities?.reduce(
+      (acc, { numberOfPacks }) => acc + numberOfPacks,
+      0
+    );
+    const difference = newVal - (newAllocatedTotal ?? 0);
+
+    if (difference > 0) {
+      const warningSnack = warning(
+        t('warning.cannot-create-placeholder', {
+          quantity: difference,
+        })
+      );
+      warningSnack();
+    }
```

Comment:
adding this line earlier in the component:
```
  const { format } = useFormatNumber();
```

and then doing this, to allow for correct calculation of the discrepancy and display of packs / units:

```suggestion
 const allocateInUnits = packSize === null;

    const newAllocatedTotal = newAllocateQuantities?.reduce(
      (acc, { numberOfPacks, packSize }) =>
        acc + numberOfPacks * (allocateInUnits ? packSize : 1),
      0
    );
    const difference = newVal - (newAllocatedTotal ?? 0);

    if (difference > 0 && newAllocatedTotal !== undefined) {
      const warningSnack = warning(
        t(
          allocateInUnits
            ? 'warning.cannot-create-placeholder-units'
            : 'warning.cannot-create-placeholder-packs',
          {
            quantity: format(newAllocatedTotal),
          }
        )
      );
      warningSnack();
    }
```